core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -354,128 +354,150 @@ object ConfigCommand extends Logging {
+    } else { // ConfigType.User
+      if (unknownConfigsToAdd.nonEmpty)
+        throw new IllegalArgumentException(s"Only quota and SCRAM credential configs can be added for '${ConfigType.USER}' using --bootstrap-server. Unexpected config names: $unknownConfigsToAdd")
+      if (unknownConfigsToDelete.nonEmpty)
+        throw new IllegalArgumentException(s"Only quota and SCRAM credential configs can be deleted for '${ConfigType.USER}' using --bootstrap-server. Unexpected config names: $unknownConfigsToDelete")
+      if (scramConfigsToAddMap.nonEmpty || scramConfigsToDelete.nonEmpty) {
+        if (entityNames.exists(_.isEmpty)) // either --entity-type users --entity-default or --user-defaults
+          throw new IllegalArgumentException("The use of --entity-default or --user-defaults is not allowed with User SCRAM Credentials using --bootstrap-server.")
+        if (hasQuotaConfigsToAdd || hasQuotaConfigsToDelete)
+          throw new IllegalArgumentException(s"Cannot alter both quota and SCRAM credential configs simultaneously for '${ConfigType.USER}' using --bootstrap-server.")
+      }
     }
 
-    if (entityNameHead.nonEmpty)
-      println(s"Completed updating config for ${entityTypeHead.dropRight(1)} $entityNameHead.")
-    else
-      println(s"Completed updating default config for $entityTypeHead in the cluster.")
+    if (hasQuotaConfigsToAdd || hasQuotaConfigsToDelete) {
+      alterQuotaConfigs(adminClient, entityTypes, entityNames, configsToBeAddedMap, configsToBeDeleted)
+    } else {
+      // handle altering user SCRAM credential configs
+      if (entityNames.size != 1)
+        // should never happen, if we get here then it is a bug
+        throw new IllegalStateException(s"Altering user SCRAM credentials should never occur for more zero or multiple users: $entityNames")
+      alterUserScramCredentialConfigs(adminClient, entityNames.head, scramConfigsToAddMap, scramConfigsToDelete)
+    }
+  }
+
+  private def alterIpConfig(adminClient: Admin, entityTypes: List[String], entityNames: List[String], configsToBeAddedMap: Predef.Map[String, String], configsToBeAdded: Predef.Map[String, ConfigEntry], configsToBeDeleted: Seq[String]) = {
+    val unknownConfigs = (configsToBeAdded.keys ++ configsToBeDeleted).filterNot(key => DynamicConfig.Ip.names.contains(key))
+    if (unknownConfigs.nonEmpty)
+      throw new IllegalArgumentException(s"Only connection quota configs can be added for '${ConfigType.IP}' using --bootstrap-server. Unexpected config names: ${unknownConfigs.mkString(",")}")
+    alterQuotaConfigs(adminClient, entityTypes, entityNames, configsToBeAddedMap, configsToBeDeleted)
+  }
+
+  private def alterClientMetricsConfig(adminClient: Admin, entityTypeHead: String, entityNames: List[String], configsToBeAdded: Predef.Map[String, ConfigEntry], configsToBeDeleted: Seq[String]) = {
+    entityNames.foreach { entityName =>
+      getOldConfig(adminClient, entityTypeHead, configsToBeDeleted, entityName)
```

Comment:
I guess you want to reuse the validation, right? However, the method name "get" is a bit weird since it is used in the "foreach". Could you please do a bit refactor?

Reply:
BTW, could you add test for `InvalidConfigurationException`?